test/test_http11.rb

```diff
@@ -183,4 +183,31 @@ def test_horrible_queries
     end
 
   end
+
+  def test_13632_workaround
+    pipes = []
+    threads = []
+    10.times do
+      r, w = IO.pipe
+      pipes << [r, w]
+      threads << Thread.new do
+        while r.gets
+        end rescue IOError
+        if Thread.current.pending_interrupt?
+          raise "Ruby bug 13632 detected for version #{RUBY_VERSION}, please open a bugreport for Puma." unless Thread.current.respond_to? :purge_interrupt_queue
```

Comment:
technically speaking `Thread.current.respond_to? :purge_interrupt_queue` is defined only for `%w(2.2.7 2.3.4 2.4.1)`, so this test should fail for all other ruby versions. What is the sense?